File: library/src/main/java/me/proxer/library/api/notifications/DeleteNotificationEndpoint.java
package me.proxer.library.api.notifications;

import me.proxer.library.api.Endpoint;
import me.proxer.library.api.ProxerCall;

/**
 * Endpoint for deleting a single or all notifications.
 *
 * @author Ruben Gees
 */
public final class DeleteNotificationEndpoint implements Endpoint<Void> {

    private final InternalApi internalApi;

    private final String id;

    DeleteNotificationEndpoint(final InternalApi internalApi, final String id) {
        this.internalApi = internalApi;
        this.id = id;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public ProxerCall<Void> build() {
        return internalApi.deleteNotification(id);
    }
}

File: library/src/main/kotlin/me/proxer/library/api/notifications/DeleteNotificationEndpoint.kt
package me.proxer.library.api.notifications

import me.proxer.library.ProxerCall
import me.proxer.library.api.Endpoint

/**
 * Endpoint for deleting a single or all notifications.
 *
 * @author Ruben Gees
 */
class DeleteNotificationEndpoint internal constructor(
    private val internalApi: InternalApi,
    private val id: String
) : Endpoint<Unit?> {

    override fun build(): ProxerCall<Unit?> {
        return internalApi.deleteNotification(id)
    }
}
